Model/Component/AdminRoles.php

```diff
@@ -0,0 +1,134 @@
+<?php
+namespace CtiDigital\Configurator\Model\Component;
+
+use Symfony\Component\Yaml\Yaml;
+use Magento\Authorization\Model\RoleFactory;
+use Magento\Framework\ObjectManagerInterface;
+use Magento\Authorization\Model\RulesFactory;
+use CtiDigital\Configurator\Model\LoggingInterface;
+use Magento\Authorization\Model\UserContextInterface;
+use Magento\Authorization\Model\Acl\Role\Group as RoleGroup;
+use CtiDigital\Configurator\Model\Exception\ComponentException;
+
+class AdminRoles extends YamlComponentAbstract
+{
+    protected $alias = 'adminroles';
+    protected $name = 'Admin Roles';
+    protected $description = 'Component to create Admin Roles';
+
+    /**
+     * RoleFactory
+     *
+     * @var roleFactory
+     */
+    protected $roleFactory;
+
+    /**
+     * RulesFactory
+     *
+     * @var rulesFactory
+     */
+    protected $rulesFactory;
+
+    /**
+     * AdminRoles constructor.
+     * @param LoggingInterface $log
+     * @param ObjectManagerInterface $objectManager
+     * @param RoleFactory $roleFactory
+     * @param RulesFactory $rulesFactory
+     */
+    public function __construct(
+        LoggingInterface $log,
+        ObjectManagerInterface $objectManager,
+        RoleFactory $roleFactory,
+        RulesFactory $rulesFactory
+    ) {
+        parent::__construct($log, $objectManager);
+
+        $this->roleFactory = $roleFactory;
+        $this->rulesFactory = $rulesFactory;
+    }
+
+    /**
+     * @param array $data
+     * @SuppressWarnings(PHPMD)
+     */
+    protected function processData($data = null)
+    {
+
+        if (isset($data['adminroles'])) {
+            foreach ($data['adminroles'] as $role) {
+                try {
+                    if (isset($role['name'])) {
+                        $this->createAdminRole($role['name'], $role['resources']);
+                    }
+                } catch (ComponentException $e) {
+                    $this->log->logError($e->getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Create Admin user roles, or update them if they exist
+     *
+     * @param $roleName
+     * @param $resources
+     */
+    private function createAdminRole($roleName, $resources)
```

Comment:
Maybe add the types of `$roleName` and `$resources` in phpdoc and the signature, too, if you can? I think for `$resources` you can, as that's an array.